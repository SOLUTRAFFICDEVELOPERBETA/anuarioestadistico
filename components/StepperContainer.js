import React from 'react';
import PropTypes from 'prop-types';
import { Button, makeStyles, Step, StepContent, StepLabel, Stepper } from '@material-ui/core';

// Estilos del componente
const useStyles = makeStyles((theme) => ({
    root: {
        width: '100%'
    },
    button: {
        marginTop: theme.spacing(1),
        marginRight: theme.spacing(1)
    },
    actionsContainer: {
        marginBottom: theme.spacing(2)
    }
}));

/**
 * Componente que permite utilizar un Stepper con configuración sencilla
 * @param {{ id: string, children: any, StepperConfig: any, onOver: () => func }} props Propiedades del componente
 */
const StepperContainer = ({ id, children, StepperConfig, onOver }) => {
    const classes = useStyles();
    const [activeStep, setActiveStep] = React.useState(0);

    // Método para avanzar en el stepper
    const handleNext = () => {
        if (activeStep === stepCount - 1) {
            onOver();
        } else {
            setActiveStep((prevActiveStep) => prevActiveStep + 1);
        }
    };

    // Método para retroceder en el stepper
    const handleBack = () => {
        setActiveStep((prevActiveStep) => prevActiveStep - 1);
    };

    const stepCount = React.Children.count(children);

    return (
        <React.Fragment>
            <div className={classes.root}>
                <Stepper activeStep={activeStep} orientation="vertical">
                    {React.Children.map(children, (child, index) => {
                        const { label = '' } = StepperConfig[index] || {};
                        return (
                            <Step key={`step-${id}-${index}`}>
                                <StepLabel onClick={() => setActiveStep(index)}>{label}</StepLabel>
                                <StepContent>
                                    {child}
                                    <div className={classes.actionsContainer}>
                                        <div>
                                            <Button
                                                disabled={activeStep === 0}
                                                onClick={handleBack}
                                                className={classes.button}>
                                                Volver
                                            </Button>
                                            <Button
                                                variant="contained"
                                                color="primary"
                                                onClick={handleNext}
                                                className={classes.button}>
                                                {activeStep === stepCount - 1
                                                    ? 'Terminar'
                                                    : 'Siguiente'}
                                            </Button>
                                        </div>
                                    </div>
                                </StepContent>
                            </Step>
                        );
                    })}
                </Stepper>
            </div>
        </React.Fragment>
    );
};

StepperContainer.propTypes = {
    id: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired,
    StepperConfig: PropTypes.array,
    onOver: PropTypes.func
};

StepperContainer.defaultProps = {
    StepperConfig: [],
    onOver: () => console.log('Fin')
};

export default StepperContainer;
