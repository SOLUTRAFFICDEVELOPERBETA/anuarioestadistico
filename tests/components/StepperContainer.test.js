import { Box, Typography } from '@material-ui/core';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import StepperContainer from '../../components/StepperContainer';

describe('Pruebas en el componente <StepperContainer/>', () => {
    const onOver = jest.fn();
    const wrapper = shallow(
        <StepperContainer
            id={`guide-help-pruebas`}
            StepperConfig={[
                { label: 'Datos' },
                
            ]}
            onOver={onOver}
        >
            <Box>
                <Typography>Datos</Typography>
            </Box>
           
            
        </StepperContainer>
    )

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });


    test('No debe llamar la función onOver()', ()=> {
        const wrapper = shallow(
            <StepperContainer
                id={`guide-help-pruebas`}
                StepperConfig={[
                    { label: 'Datos' },
                    { label: 'Tipos' },
                    
                ]}
                onOver={onOver}
            >
                <Box>
                    <Typography>Datos</Typography>
                </Box>
                <Box>
                    <Typography>tipos</Typography>
                </Box>
               
                
            </StepperContainer>
        )
        wrapper.find('WithStyles(ForwardRef(Button))').at(1).prop('onClick')();
        expect(onOver).not.toHaveBeenCalled();
        
    })


    test('debe llamar la función onOver()', () => {
        wrapper.find('WithStyles(ForwardRef(Button))').at(1).prop('onClick')();
        expect(onOver).toHaveBeenCalled();
        expect(onOver).toHaveBeenCalledTimes(1);
    })

   
})