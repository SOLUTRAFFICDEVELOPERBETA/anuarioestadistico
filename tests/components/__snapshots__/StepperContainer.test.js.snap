// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <StepperContainer/> debe mostrarse correctamente 1`] = `
<Fragment>
  <div
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(Stepper))
      activeStep={0}
      orientation="vertical"
    >
      <WithStyles(ForwardRef(Step))
        key="step-guide-help-pruebas-0/.0"
      >
        <WithStyles(ForwardRef(StepLabel))
          onClick={[Function]}
        >
          Datos
        </WithStyles(ForwardRef(StepLabel))>
        <WithStyles(ForwardRef(StepContent))>
          <Styled(MuiBox)>
            <WithStyles(ForwardRef(Typography))>
              Datos
            </WithStyles(ForwardRef(Typography))>
          </Styled(MuiBox)>
          <div
            className="makeStyles-actionsContainer-3"
          >
            <div>
              <WithStyles(ForwardRef(Button))
                className="makeStyles-button-2"
                disabled={true}
                onClick={[Function]}
              >
                Volver
              </WithStyles(ForwardRef(Button))>
              <WithStyles(ForwardRef(Button))
                className="makeStyles-button-2"
                color="primary"
                onClick={[Function]}
                variant="contained"
              >
                Terminar
              </WithStyles(ForwardRef(Button))>
            </div>
          </div>
        </WithStyles(ForwardRef(StepContent))>
      </WithStyles(ForwardRef(Step))>
    </WithStyles(ForwardRef(Stepper))>
  </div>
</Fragment>
`;
